### PR TITLE
Change playground options layout

### PR DESCRIPTION
**Pull request by duailibe**
Context: https://github.com/prettier/prettier/issues/3352#issuecomment-348387918

Preview: https://deploy-preview-3374--prettier.netlify.com/playground

### CODE REVIEW

**Comment by duailibe** [comment]
Actually for now I think I'd prefer a :+1:/:-1: on the general layout, because I think the options layout can use some love (hiding some options depending on the parser) and I think these extensions would be great for a React-based site.

File: website/pages/playground/index.html [hunk 1]
```diff
@@ -339,58 +357,14 @@ <h1>Prettier <span id="version"></span></h1>
 
   <div class="bottom-bar">
     <div class="bottom-bar-buttons">
+      <button type="button" class="btn" id="button-options">
+        See options
```
Comment (ikatyang) [inline website/pages/playground/index.html]: `Show options`?

**Comment by ikatyang** [comment]
It'd be better to categorize options, but it should be in another PR.

- Global
  - printWidth
  - tabWidth
  - useTabs
  - parser
- JavaScript
  - semi
  - singleQuote
  - bracketSpacing
  - jsxBracketSameLine
  - trailingComma
- Markdown
  - proseWrap
- Special
  - rangeStart
  - rangeEnd
  - insertPragma
  - requirePragma
- Debug
  - show AST
  - show doc
  - show second format

**Comment by lipis** [comment]
Is there a blocker on this one?! Can we have that in time for 1.9?

File: website/pages/playground/index.html [hunk 2]
```diff
@@ -192,47 +189,27 @@
       }
     }
 
-    .options-details {
-      padding: 8px 0;
+    .options-container {
+      max-height: 100%;
+      overflow-y: scroll;
```
Comment (lydell) [inline website/pages/playground/index.html]: Could you change this to `overflow-y: auto;`. It looks strange to have a disabled scroll bar when there’s no overflow (that’s what it looks like on Windows and Linux).

File: website/pages/playground/index.html (hunk 1 above)
Comment (lipis) [reply website/pages/playground/index.html]: That copy is better.. let's change that as well.

**Comment by duailibe** [comment]
@lipis there’s no need to rush with this, not related to 1.9 at all. I’m not sure we shoud merge as-is yet, current layout has better options separation

File: website/pages/playground/index.html (hunk 2 above)
Comment (duailibe) [reply website/pages/playground/index.html]: Yeah I do, thanks for noticing

File: website/pages/playground/index.html [hunk 3]
```diff
@@ -192,47 +189,45 @@
...
+      padding-bottom: 5px;
+    }
+
+    .sub-options > * {
+      margin-left: 10px;
+    }
+
+    label {
+      font-family: "Consolas", "Courier New", Courier, monospace;
+      padding: 10px 0;
```
Comment (ikatyang) [inline website/pages/playground/index.html]:
```diff
-padding: 10px 0;
+margin: 10px 0;
+display: block;
```

image

image

File: website/pages/playground/index.html [hunk 4]
```diff
@@ -382,27 +356,63 @@ <h1>Prettier <span id="version"></span></h1>
...
+
+          <label><input type="checkbox" id="insertPragma"> --insert-pragma</label>
+          <label><input type="checkbox" id="requirePragma"> --require-pragma</label>
+        </details>
+        <details class="sub-options">
+          <summary>Debug</summary>
+
           <label><input type="checkbox" id="ast"> show AST (debug)</label>
           <label><input type="checkbox" id="doc"> show doc (debug)</label>
           <label><input type="checkbox" id="output2"> show second format (debug)</label>
```
Comment (ikatyang) [inline website/pages/playground/index.html]: I think we don't need `(debug)` anymore :)

File: website/pages/playground/index.html [hunk 5]
```diff
@@ -318,57 +313,36 @@ <h1>Prettier <span id="version"></span></h1>
...
-      </button>
-    </div>
-
-    <details class="options-details">
-      <summary class="options-summary">Options</summary>
+    <details id="options-details" class="options-container">
+      <summary class="options-summary"></summary>
+      <div class="options">
+        <details class="sub-options">
+          <summary>Global</summary>
```
Comment (ikatyang) [inline website/pages/playground/index.html]: It'd be better to open (`<details open>`) `Global` and `JavaScript` by default I think.

Comment (duailibe) [reply website/pages/playground/index.html]: Yeah that could work.. I'd rather persist the last state though.

(That'd be so easy to in a react component 😅)

**Comment by lydell** [comment]
@duailibe Now the scrollbar only appears if needed, great! However, the whole page still scrolls instead of just the sidebar in Firefox.

**Comment by duailibe** [comment]
Now tested on chrome, firefox and safari on mac :-)

**Comment by azz** [comment]
The show options button doesn't seem to work on my phone (S8 Chrome)

**Comment by azz** [comment]
Wait now it does - bad caching

**Comment by duailibe** [comment]
I was just gonna ask if you could hard reload on the phone 😅

**Pull request merged**